app/src/main/java/org/mozilla/focus/experiments/NimbusSetup.kt

```diff
@@ -0,0 +1,97 @@
+
+        // The name "focus-android" or "klar-android" here corresponds to the app_name defined
+        // for the family of apps that encompasses all of the channels for the Focus app.
+        // This is defined upstream in the telemetry system. For more context on where the
+        // app_name come from see:
+        // https://probeinfo.telemetry.mozilla.org/v2/glean/app-listings
+        // and
+        // https://github.com/mozilla/probe-scraper/blob/master/repositories.yaml
+        val appInfo = NimbusAppInfo(
+            appName = getNimbusAppName(),
+            // Note: Using BuildConfig.BUILD_TYPE is important here so that it matches the value
+            // passed into Glean. `Config.channel.toString()` turned out to be non-deterministic
+            // and would mostly produce the value `Beta` and rarely would produce `beta`.
+            channel = BuildConfig.BUILD_TYPE
+        )
+        Nimbus(context, appInfo, serverSettings).apply {
+            // This performs the minimal amount of work required to load branch and enrolment data
+            // into memory. If `getExperimentBranch` is called from another thread between here
+            // and the next nimbus disk write (setting `globalUserParticipation` or
+            // `applyPendingExperiments()`) then this has you covered.
+            // This call does its work on the db thread.
+            initialize()
+
+            if (!enabled) {
+                // This opts out of nimbus experiments. It involves writing to disk, so does its
+                // work on the db thread.
+                globalUserParticipation = enabled
+            }
+
+            if (url.isNullOrBlank()) {
+                setExperimentsLocally(R.raw.initial_experiments)
+            }
+
+            // We may have downloaded experiments on a previous run, so let's start using them
+            // now. We didn't do this earlier, so as to make getExperimentBranch and friends returns
+            // the same thing throughout the session. This call does its work on the db thread.
+            applyPendingExperiments()
+
+            // Now fetch the experiments from the server. These will be available for feature
+            // configuration on the next run of the app. This call launches on the fetch thread.
+            fetchExperiments()
+        }
+    } catch (e: Throwable) {
+        // Something went wrong. We'd like not to, but stability of the app is more important than
+        // failing fast here.
+        val crashReporter = context.components.crashReporter
+        if (crashReporter.enabled) {
+            crashReporter.submitCaughtException(e)
+        } else {
+            Logger.error("Failed to initialize Nimbus", e)
+        }
+        NimbusDisabled()
+    }
+
+fun getNimbusAppName(): String {
+    return if (BuildConfig.FLAVOR.contains("focus")) {
+        "focus-android"
+    } else {
+        "klar-android"
+    }
```

Comment:
These are likely to be the correct names, but I think that `_` rather than `-` will be used.  See here: https://github.com/mozilla/probe-scraper/blob/edd24dda40487f0274da3182a58baf8d02f0582e/repositories.yaml#L499

Reply:
Meaning, I expect `focus_android` and `klar_android`.

Reply:
Oops, underscore is what I meant to have used. Thanks for the catch!